### PR TITLE
JS: Fix duplicated `prettier-ignore` comments

### LINKED ISSUE #10661: Comment is duplicated when prettier-ignore is used
**Prettier 2.2.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDAdAIwDpQAR4D0heADgE5wwwCWc5AtDQOZQSW4HF4DKEAtnDwQYAC3p5I-QbDziO+POjAgANCAila0AM7JQqcuQgB3AAqGEelKgA2J1AE896zOVRgA1lR6kPNKGZkGHIAVzh1OH5MOAATWLiAGVRA0NRmOAAxdn5UagCgm1CYCDUQURh+WwB1URp4HT8wOB4repoAN3rHZHAdFxAAnXoYM3dmXOQAMzth9QArHQAPACF3Lx9UQUSAuGnZiJBFpZ4C2zgARVCRPaQZ2zmQP3Jh8l7MVBjbMooAmGqaLExMgABwABnUFAgw2q7lIvQocFeHT26gAjtd4GNNNY0DoGFA4HE4mVKBiaJQxulJncDuphvwaMEwocdGdLpjbvdHjBPgCgaJkAAmdQhVA0WwFADCAhpICRAFYyqFhgAVT7WbmHDrhACSUASsB4YHINC0AEEDTwYI5zvsHnAAL6OoA)

```sh
--parser babel
```

**Input:**

```jsx
verylongidentifierthatwillwrap123123123123123(
  a.b
    // prettier-ignore
    // Some other comment here
    .c
);
```

**Output:**

```jsx
verylongidentifierthatwillwrap123123123123123(
  // prettier-ignore
  // Some other comment here
  a.b
    // prettier-ignore
    // Some other comment here
    .c
);
```

If you feed the output into Prettier again, it duplicates the comment again:

**Input:**

```jsx
verylongidentifierthatwillwrap123123123123123(
  // prettier-ignore
  // Some other comment here
  a.b
    // prettier-ignore
    // Some other comment here
    .c
);
```

**Output:**

```jsx
verylongidentifierthatwillwrap123123123123123(
  // prettier-ignore
  // Some other comment here
  // prettier-ignore
  // Some other comment here
  a.b
    // prettier-ignore
    // Some other comment here
    .c
);
```

This means that every time I save the file, the comment gets duplicated again above the node 😄

**Expected behavior:**
Comment should not be duplicated

### PR DESCRIPTION
**Pull request by fisker**
## Description

Fixes #10661

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by thorn0** [comment]
Can we skip printing comments that are marked as `printed`? Did you consider this solution?

**Comment by fisker** [comment]
@thorn0 I tried that long time ago, comments not only printed once, so we can't do that.

**Comment by fisker** [comment]
I guess removing comments failed for the same reason, some node is printed multiple times.

**Comment by thorn0** [comment]
Rather because the condition `hasComment(node, CommentCheckFlags.PrettierIgnore)` doesn't hold after the comment is removed.

File: src/main/comments.js [hunk 1]
```diff
@@ -555,6 +561,10 @@ function printCommentsSeparately(path, options) {
 
   path.each((commentPath) => {
     const comment = commentPath.getValue();
+    if (skip && skip.has(comment)) {
+      return;
```
Comment (thorn0) [inline src/main/comments.js:565]: Looks good. Can you add a test to cover this line?

File: src/main/comments.js [hunk 2]
```diff
@@ -540,12 +539,19 @@ function printDanglingComments(path, options, sameIndent, filter) {
   return indent([hardline, join(hardline, parts)]);
 }
 
-function printCommentsSeparately(path, options) {
+function printCommentsSeparately(path, options, skip) {
```
Comment (thorn0) [inline src/main/comments.js:542]: `skip` is a verb. Only functions should be named with verbs. How about `ignored`?

**Review by thorn0 — APPROVED** [review]

**Pull request merged**